students/AurelPerianu/Lesson9/mailroom_5.py

```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/env python3
+
+
+class Donor:
+
+    def __init__(self, name, donations=None):
+        if ' ' in name:
+            self._first, self._last = name.split(' ')
+        else:
+            self._first = ''
+            self._last = name
+        self._name = name
+        if donations:
+            self._donations = donations
+        else:
+            self._donations = []
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def first_name(self):
+        return self._first
+
+    @property
+    def last_name(self):
+        return self._last
+
+    @property
+    def donations(self):
+        return self._donations
+
+    @property
+    def full_name(self):
+        return f"{self._first} {self._last}"
+
+    def __str__(self):
+        return f"{self._first} {self._last} {self._donations}"
+
+    def __repr__(self):
+        return f'Donor(\'{str(self._name)}\', {self._donations})'
+
+    def add_donations(self, amount):
+        return self._donations.append(amount)
+
+    @property
+    def total_donations(self):
+        return sum(self._donations)
+
+    @property
+    def len_donations(self):
+        return len(self._donations)
+
+    @property
+    def avg_donations(self):
+        try:
+            return self.total_donations / self.len_donations
+        except ValueError:
+            return self._donations
+
+    def letter(self, amount):
+        txt = """\n
+            To:       {0:s}
+            Subject:  Your donation of ${1:,.2f}
+            Dear {0:s},\n
+
+            Thank you for your donation of ${1:,.2f}.\n
+            """
+        print(txt.format(self._name, amount))
+
+
+class Donors_All:
+    def __init__(self, donors=None):
+        if donors:
+            self._donors = donors
+        else:
+            self._donors = []
```

Comment:
you can also use shortcuts:
`self.donors = donors or []`
or
`self.donors = donors if donors else []`